### PR TITLE
Progress some planned deprecations

### PR DESCRIPTION
**Pull request by MichaelChirico**
#5840 will be filed separately since it's a new deprecation --> warrants its own PR.

These are simply progressing existing deprecation cycles to their next steps.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -609,5 +609,10 @@
 
 15. Thanks to @ssh352, Václav Tlapák, Cole Miller, András Svraka and Toby Dylan Hocking for reporting and bisecting a significant performance regression in dev. This was fixed before release thanks to a PR by Jan Gorecki, [#5463](https://github.com/Rdatatable/data.table/pull/5463). 
 
+16. `key(x) <- value` is now fully deprecated (from warning to error). Use `setkey()` to set a table's key. We started warning not to use this approach in 2012, with a stronger warning starting in 2019 (1.12.2). This function will be removed in the next release.
+
+17. Argument `logicalAsInt` to `fwrite()` now warns. Use `logical01` instead. We stated the intention to begin removing this option in 2018 (v1.11.0). It will be upgraded to an error in the next release before being removed in the subsequent release.
```
Comment (jangorecki) [inline NEWS.md:614]: Possibly introduces revdeps failures, and slows down release by asking them to amend.

**Review by jangorecki — COMMENTED** [review]
logicalAsInt we could move post release so revdeps (and non cran pkgs) have plenty of time to amend their code. Other looks good

File: NEWS.md (hunk 1 above)
Comment (MichaelChirico) [reply NEWS.md:614]: I disagree. Once again we should merge and check revdeps.

Moreover, in this case we can trivially check the entirety of GitHub for downstream usages, of which there are practically 0 -- most of the 41 results here are just {data.table} forks/copies:

https://github.com/search?q=lang%3AR%20%2FlogicalAsInt%5Cs*%3D%2F&type=code

Comment (MichaelChirico) [reply NEWS.md:614]: There are no (0) CRAN packages using `logicalAsInt`:

https://github.com/search?q=lang%3AR+%2FlogicalAsInt%5Cs*%3D%2F+org%3Acran&type=code

**Review by jangorecki — APPROVED** [review]

**Comment by MichaelChirico** [comment]
Refined the GitHub search and confirmed there's only two places using logicalAsInt, neither of which have any activity for >5 years. Nevertheless filed downstream issues about the pending switch.

**Comment by jangorecki** [comment]
If packages were not active for 5 years, then it is more likely they will not manage to update their code within short period of 2 weeks that CRAN gives. Therefore we are risking this change to be reason for archival of two packages. I would say we should try yo avoid doing so. Giving them time of full 1.15.99 cycle doesn't guarantee they will be fixed but at least feels more appropriate, as one of the goals of DT, even officially defined recently, is backward compatibility.

**Comment by MichaelChirico** [comment]
> If packages were not active for 5 years, then it is more likely they will not manage to update their code within short period of 2 weeks that CRAN gives. Therefore we are risking this change to be reason for archival of two packages. I would say we should try to avoid doing so. Giving them time of full 1.15.99 cycle doesn't guarantee they will be fixed but at least feels more appropriate, as one of the goals of DT, even officially defined recently, is backward compatibility.

to be clear these are not CRAN packages, they are only maintained on GitHub.

**Comment by jangorecki** [comment]
It feels much safer then, yet not ideal.

**Pull request merged**